src/components/spatial/Spatial.js

```diff
@@ -174,36 +191,42 @@ export default class Spatial extends AbstractSelectableComponent {
     });
   }
 
-  renderImages(viewProps) {
-    if (!this.props.images) {
-      return null;
-    }
-    if (this.props.clearPleaseWait) {
-      this.props.clearPleaseWait('images');
-    }
-    const imageNames = Object.keys(this.props.images).reverse();
-    // We want the z-order to be the opposite of the order listed.
-    const visibleImageNames = imageNames.filter(name => this.state.layerIsVisible[name]);
-    const visibleImages = visibleImageNames.map(name => this.props.images[name]);
-    const tileSources = visibleImages.map(image => image.tileSource);
-    const samples = visibleImages.map(image => image.sample || 1);
-    const translateXs = visibleImages.map(image => image.translateX || 0);
-    const translateYs = visibleImages.map(image => image.translateY || 0);
-    if (!samples.every(sample => sample === samples[0])
-      && !translateXs.every(x => x === translateXs[0])
-      && !translateYs.every(y => y === translateYs[0])) {
-      // This is easiest for now: if the data changes, we can change.
-      throw new Error(`sample, translateX, and translateY must match for all images: ${this.props.images}`);
-    }
-    return (
-      <OpenSeadragonComponent
-        tileSources={tileSources}
-        sample={samples[0]}
-        translateX={translateXs[0]}
-        translateY={translateYs[0]}
-        {...viewProps}
-      />
-    );
+  renderImagesLayer() {
+    const layers = [];
+    this.images.forEach((layer) => { layers.push(this.createTileLayer(layer)); });
+    return layers;
+  }
+
+  createTileLayer(layer) {
+    const [layerType, source] = layer;
+    return new IdentityCoordinatesTileLayer({
+      id: `${layerType}-tile-layer`,
+      pickable: true,
+      coordinateSystem: COORDINATE_SYSTEM.IDENTITY,
+      maxCacheSize: 50,
+      getTileData: ({ x, y, z }) => loadImage(`${source.tileSource}/${layerType}_files/${z + 16}/${x}_${y}.jpeg`),
```

Comment:
Explain where 16 and 50 come from... The 16 depends on our particular data set and shouldn't be hardcoded, right?

Reply:
Yes - the 50 was also somewhat random just to get decent performance.  I'll probably remove in favor of the default then as long as the performance isn't so bad